src/kademlia/get_item.cpp

```diff
@@ -78,7 +88,7 @@ void get_item::got_data(bdecode_node const& v,
 		// and it's immutable
 
 		m_data.assign(v);
-		bool put_requested = m_data_callback(m_data);
+		bool put_requested = m_data_callback(m_data, false);
```

Comment:
I think it will only be true at done(). 
And for immutable data, the callback function will ignore the second parameter.